modules/persistent-storage-csi-snapshots-delete.adoc

```diff
@@ -24,6 +21,67 @@ metadata:
 driver: hostpath.csi.k8s.io
 deletionPolicy: Delete <1>
 ----
-<1> If the `Delete` value is set, the underlying snapshot will be deleted, along with the `VolumeSnapshotContent` object. If the `Retain` value is set, both the underlying snapshot and `VolumeSnapshotContent` object remain.
+<1> When deleting the volume snapshot, if the `Delete` value is set, the underlying snapshot will be deleted, along with the `VolumeSnapshotContent` object. If the `Retain` value is set, both the underlying snapshot and `VolumeSnapshotContent` object remain.
   +
-If the `Retain` value is set, and the `VolumeSnapshot` object is deleted without deleting the corresponding `VolumeSnapshotContent` object, then the content will remain. The snapshot itself is also retained in the storage back end.
+If the `Retain` value is set and the `VolumeSnapshot` object is deleted without deleting the corresponding `VolumeSnapshotContent` object, the content will remain. The snapshot itself is also retained in the storage back end.
+
+. Delete the volume snapshot:
+
+.. If the deletion policy is set to `Delete`, enter the following command:
++
+[source,terminal]
+----
+$ oc delete volumesnapshot <volumesnapshot_name>
+----
++
+.Example output
+[source, terminal]
+----
+volumesnapshot.snapshot.storage.k8s.io "mysnapshot" deleted
+----
+.. If the deletion policy is set to `Retain` and you want to delete the volume snapshot content, enter the following commands:
++
+[source,terminal]
+----
+$ oc delete volumesnapshot <volumesnapshot_name>
+----
++
+[source,terminal]
+----
+$ oc delete volumesnapshotcontent <volumesnapshotcontent_name>
+----
++
+. Optional: If the `VolumeSnapshot` object is not successfully deleted, enter the following commands to remove the existing finalizers from the `VolumeSnapshot` object so that the delete operation can continue:
++
+[IMPORTANT]
+====
+Only remove the finalizers if you are confident that there are no existing references from either persistent volume claims or volume snapshot contents to the `VolumeSnapshot` object.
+
+Even with the `--force` option, the delete operation does not delete snapshot objects until all finalizers are removed.
+====
+
+.. Remove the finalizer for the leftover resources by patching the finalizer with null:
```

Comment:
While accurate, I think this would be cleaner to say:

> Remove any finalizers for the leftover resource:

We're not technically patching the finalizer, we're patching the VolumeSnapshot object to have it's finalizers removed. Your sentence is accurate, but it feels clunky to me.

Reply:
Makes sense, I rearranged this step slightly to simplify and use your wording.